[LIN ZHIWEI] iP

Zzhiwei opened the pull request:
# IP pull request

* hello
* how are you
* i am under the water

1. eat
2. sleep
3. code
4. repeat

```C 
int main() {
    int a = 3;
    int *p = &a;
    return 1;
}
```

Things to consider:
- [ ] address book javadocs class has but methods don't have
- [ ] javadocs for tests?
- [ ] cs2102 t2 
- [ ] cs2103T assignments

:grinning: :zany_face: :stuck_out_tongue_closed_eyes:

Albert Einstein once said: 
> Just because someone quoted someone famous saying something on the internet, don't believe it is true.

http://google.com
www.facebook.com

This is a piece of inline code: `var x = 3;`

*Hello*, I am **Zhiwei**, currently studying ~~CS~~.

Commit:
Add Gradle support

Zzhiwei committed:
Level Completion: Level 1

Completed level 1.

Zzhiwei committed:
Level completion: complete level 2

Completed level 2.

Zzhiwei committed:
Level completion: complete level 3

Completed level 3.

Zzhiwei committed:
Level completion: complete level 4

Completed level 4.

Zzhiwei committed:
Level completion: complete level 5

Completed level 5.

Zzhiwei committed:
Level completion: complete level 5

Completed level 5.

Zzhiwei committed:
Level completion: complete level 6

Completed level 6.

Zzhiwei committed:
Finish half of level 7

Zzhiwei committed:
Level completion: finish level 7

Zzhiwei committed:
Level completion: finish level 8

Make Yoyo recognise datetime.

Zzhiwei committed:
Merge pull request #1 from softmagnet/branch-Level-7

Merge branch level 7 into master

Zzhiwei committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Event.java
#	src/main/java/Yoyo.java

Zzhiwei committed:
A-MoreOOP: add UI class

Zzhiwei committed:
A-MoreOOP: add TaskList class

Zzhiwei committed:
A-MoreOOP: add Storage class

Zzhiwei committed:
A_MoreOOP: finish most of OOP task

Zzhiwei committed:
A_MoreOOP: add Command class and subclasses

Zzhiwei committed:
Level completion: complete A_MoreOOP tasks

Zzhiwei committed:
Merge branch 'OOP'

Zzhiwei committed:
Level completion: finish A-Packages

Zzhiwei committed:
Test: add parser test

Zzhiwei committed:
Test: add storage test

Zzhiwei committed:
Test: add task test

Zzhiwei committed:
Fix wrong name

Change "Duke.yoyo" to "Yoyo" in Parse class

Zzhiwei committed:
A-JavaDoc: add java docs to code

Zzhiwei committed:
A-CodingStandard: modify code to meet coding standard

Zzhiwei committed:
Complete level 9 by adding find function

Zzhiwei committed:
Merge branch 'branch-Level-9'

Zzhiwei committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/yoyo/core/Parser.java

Zzhiwei committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/yoyo/Yoyo.java
#	src/main/java/yoyo/command/CommandBye.java
#	src/main/java/yoyo/command/CommandList.java
#	src/main/java/yoyo/core/Ui.java

Zzhiwei committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

Zzhiwei committed:
Add gradle to project

Zzhiwei committed:
Merge branch 'branch-A-Gradle'

Zzhiwei committed:
Use checkstyle in gradle to correct coding standard violations

Zzhiwei committed:
Merge branch 'branch-A-Checkstyle'

Zzhiwei committed:
Add GUI with main layout including images and echoing

Zzhiwei committed:
Add fxml to project and hook up Yoyo logic with GUI

szelongq commented on src/main/java/yoyo/core/Storage.java:
```diff
@@ -0,0 +1,124 @@
+                currLine = s.nextLine();
+                isCurrTaskComplete = currLine.charAt(ISDONE_STR_INDEX) == 'X';
+                char typeChar = currLine.charAt(TYPE_STR_INDEX);
+                currType = typeChar == 'T'
+                        ? TaskType.TODO
+                        : typeChar == 'D'
+                        ? TaskType.DEADLINE
+                        : TaskType.EVENT;
```
While I think this works great and is really concise, maybe you could consider using a switch expression to make it more intuitive to readers?

szelongq commented on src/main/java/yoyo/core/Parser.java:
```diff
@@ -0,0 +1,99 @@
+        String[] dateTimeArr = timeString.split(" ");
+        Command.checkCompleteCommand(dateTimeArr);
+        String[] dateArr = dateTimeArr[0].split(String.valueOf(separator));
+        if (dateArr.length != 3 || dateTimeArr[1].length() < 3) {
+            throw new YoyoException.YoyoInvalidFormatException("Invalid datetime format, "
+                    + "use 'yyyy/MM/dd hhmm'");
+        }
```
Perhaps you could consider adding a comment here or make it more explicit on how the conditions lead to an invalid datetime format?

szelongq commented on src/main/java/yoyo/command/Command.java:
```diff
@@ -0,0 +1,54 @@
+     * @param tasks Tasks currently in the Yoyo program.
+     * @param storage Storage instance of the Yoyo program.
+     * @param ui Ui instance of Yoyo program.
+     * @throws YoyoException
```
Perhaps to be consistent with your previous header comment, more information on when this exception would be thrown could be given to assist the reader in understanding the method?

szelongq reviewed (commented):
LGTM! :D
Really, really clean code! I just made a few small suggestions.
Really great work!

Zzhiwei committed:
Finish level 10 by integrating GUI

Zzhiwei committed:
Add docs for dialog and change duke to yoyo

Zzhiwei committed:
Merge branch 'branch-level-10'

Zzhiwei committed:
Add fx:root construct to main window and refactor Main.java

Zzhiwei committed:
Add exit action for "bye" command and add title to window

CraveToCode commented on src/main/java/yoyo/Yoyo.java:
```diff
@@ -0,0 +1,38 @@
+        }
+    }
+
+    public String getResponse(String input) {
```
Perhaps you could add a header JavaDoc for getResponse since it is a public method, to fit with coding style standards.

I noticed there were also a few more instances of public constructors/classes/methods without header comments, that you could perhaps take a look at too!

CraveToCode commented on src/main/java/yoyo/command/Command.java:
```diff
@@ -0,0 +1,55 @@
+import yoyo.task.TaskList;
+
+
+public abstract class Command {
```
Perhaps you could add a JavaDoc header comment for this class since it's a public class, to better fit with coding style conventions.

CraveToCode commented on src/main/java/yoyo/command/CommandDeadline.java:
```diff
@@ -0,0 +1,43 @@
+     * @param storage Storage instance of the Yoyo program.
+     * @param dialogHandler Ui instance of Yoyo program.
+     * @throws YoyoException
+     * @return
```
There is a missing JavaDoc here for return value, and I noticed there were a few more similar cases in the other Command sub-classes.

CraveToCode commented on src/main/java/yoyo/command/CommandList.java:
```diff
@@ -0,0 +1,26 @@
+    /**
+     * Executes "list" command.
+     *
+     * @param tasks   Tasks currently in the Yoyo program.
+     * @param storage Storage instance of the Yoyo program.
+     * @param dialogHandler      Ui instance of Yoyo program.
```
The spacing after the @param tasks and @param dialogHandler could be decreased to better fit with the code style standards.
There are a few more such instances across the other classes that you might want to take a look at.

CraveToCode commented on src/main/java/yoyo/core/DialogHandler.java:
```diff
@@ -0,0 +1,98 @@
+    }
+
+    /**
+     * prints success message for adding task.
```
I believe the first letter should be capitalized for sentences in JavaDoc headers.

CraveToCode reviewed (commented):
LGTM! 👍 
Overall well written, just a few recommendations to improve code style standard.

Zzhiwei committed:
Make corrections regarding PR comments and add border-radius to image

Zzhiwei committed:
Add assert statements to project

Program has no assert statements.

Adding assert statements can help check for programming errors.

Let's:
* add assert statement where it can help us identify potential programming errors

Zzhiwei committed:
Improve code quality

Certain parts of the program do not obey the code quality guideline in the textbook

By improving code quality, we can improve readability, maintainability, extensibility and robustness.

Let's:
* reduce the size of some methods by method extraction
* remove all magic literals
* remove complicated expression
* etc.

Zzhiwei committed:
Merge pull request #3 from softmagnet/A-CodeQuality

Improve code quality

Zzhiwei committed:
Merge branch 'master' of https://github.com/softmagnet/ip

Zzhiwei committed:
Merge branch 'master' into A-Assertions

# Conflicts:
#	src/main/java/yoyo/command/CommandFind.java
#	src/main/java/yoyo/core/DialogHandler.java
#	src/main/java/yoyo/core/Storage.java

Zzhiwei committed:
Merge pull request #4 from softmagnet/A-Assertions

Add assert statements

Zzhiwei committed:
Set up command for tagging (work in progress)

Zzhiwei committed:
Add tags feature to Yoyo

Yoyo does not have tagging feature.

Adding tags feature can be good if users want to categorise tasks by adding tags.

Let's:
* add tags feature

Zzhiwei committed:
Improve documentation and change style such that checkstyle passes

Zzhiwei committed:
Merge branch 'A-Tags'

Zzhiwei committed:
Add project screenshot

Zzhiwei committed:
Update README.md

Zzhiwei committed:
Update doc/README.md

Zzhiwei committed:
Delete extraneous hashtags from doc/README.md

Zzhiwei committed:
Add more tests to Yoyo software

Zzhiwei committed:
Fix font resizing problem and change label component to text

Zzhiwei committed:
Change button font family

Zzhiwei committed:
correct grammar for documentation on line 13 in DialogHandler.java

Zzhiwei committed:
Add yml file for CI

Zzhiwei committed:
Revert "Add yml file for CI"

This reverts commit c17b8eea057c3dae122679a1a1e03fe76ae6eff2.

Zzhiwei committed:
Add workflow

Zzhiwei committed:
Remove io redirection test

Zzhiwei committed:
Add space between arguments for method "getYoyoDialog"

Zzhiwei committed:
Update java workflow

Zzhiwei committed:
Update README.md
